redfish-core/include/utils/pcie_util.hpp

```diff
@@ -24,27 +24,90 @@ namespace pcie_util
 {
 
 /**
- * @brief Populate the PCIe Device list from a GetSubTreePaths search of
- *        inventory
+ * @brief Workaround to handle duplicate PCI device list
  *
- * @param[i,o] asyncResp  Async response object
- * @param[i]   Name   Key to store the list of PCIe devices in asyncResp
+ * retrieve PCI device endpoint information and if path is
+ * ~/chassisN/io_moduleN/slotN/adapterN then, replace redfish
+ * PCI device as "chassisN_io_moduleN_slotN_adapterN"
  *
- * @return void
+ * @param[i]   fullPath  object path of PCIe device
+ *
+ * @return string: unique PCIe device name
  */
+inline std::string buildPCIeUniquePath(const std::string& fullPath)
+{
+    sdbusplus::message::object_path path(fullPath);
+    sdbusplus::message::object_path parentPath = path.parent_path();
+    sdbusplus::message::object_path parentofParentPath =
+        parentPath.parent_path();
+    std::string devName;
+
```

Comment:
Looks like this is making an assumption that the PCI device endpoint would be at most three levels deep. Is that a safe assumption?

Reply:
I haven't actually read the code `buildPCIeUniquePath()` which has been inherited for long.
Reading the code, I think it should be okay  even if its depth is less than 3. It that case, it will be return "".
However,  I think this is used only for PCIeDevice which will be more than 3 depths....

Reply:
I modified it to go thru 3 depths